### ISSUE TITLE
Not an Issue: just contact

### ISSUE THREAD

**Issue by chsims1** [Saiyato/volumio-snapcast-plugin#78]
Hi Saiyato, firstly congrats on your plugin (and all the rest): it's gone a long way forward to demonstrating how this could be directly implemented in Volumio (IMHO the biggest thing missing from Volumio). 

I did try a long time a go to install SnapCast from Github, and it worked really well.  I have tried on a couple of occasions to use your plugin, but have failed each time for probably different reasons (last time was that the sync was poor).  Please take my following comments as constructive, rather than negative ;)  Firstly, it is way too complex in the number of options offered (yes, it can be complex, but there should definitely be a simple case scenario that can be set up with minimum choice).  Secondly (this could be a language problem), I wasn't always sure exactly what each option was doing.  I understand, at a reasonable level, what snapcast is doing, but I had trouble getting it set up correctly.

What I am wondering is if you would like to try and refine and make your your plugin more user friendly, or you are happy with the amount of work you have obviously already invested?  I would totally understand if you have moved on from this project. I'm very happy to help, test etc., but my javascript skills are pretty basic. I've wondered about this for a while, thinking that Mi@Volumio would incorporate it soon, but time is advancing & I think Volumio is missing a big crowd-pleaser. 

Cheers regardless, and keep on with the plugins,

Ian Sutherland

**Comment by Saiyato** [comment]
Hi Ian,

Thanks for reaching out, I've been quite pre-occupied the last two-and-a-half years on personal matters, but I do appreciate the feedback. To be honest, I've been playing with the idea for some time now, as all the options raise more questions than solve them. ;) Apart from the personal matters, I've been trying to maintain the other plugins, which takes time. As of late, I've solved a few other (quite bothering) issues for e.g. the Kodi plugin, so I'm back at SnapCast (it's his turn again :P ).

As you might (or might not) have seen, I've been working on containerized solutions for snapcast, that way I'm way less operating system dependent. Another development is splitting the client and server (for obvious reasons, why must you install a server on all clients?). While Docker might not necessarily be the best of options, it's the one I'm starting to understand (and it pairs perfectly with pydPiper, which already depends on Docker). I/we/someone just needs to be able to benchmark all of this, so I can update the documentation accordingly. I'll be more than happy to create a v0.15.0 for those who don't want Docker, but I won't be maintaining it regularly.

So yes, _a snapcast plugin_ is still relevant, but I'm starting to think I need a clean sheet, instead of applying fix, over fix, over fix, over fix. In any case, I've setup automated builds on Docker Hub (which means you will always be able to get yourself a newer or custom build, with minimal hassle). See my Docker Hub for [snapserver](https://hub.docker.com/repository/docker/saiyato/snapserver) (the client needs some work, so what you see for the server is the end game -> build per arch). This also solves another problem, multi-platform, this way I can build the same container for x86, x64, armv6, armv7, armv8 etc. etc.

So in short I will be addressing the following:
- multi-architecture
- containerize the solution (packing dependencies)
- simplifying options (but allowing for advanced config)
- solving some problems that have been there for some time (Airplay, the new Volspotconnect config, hopefully propagating track info to the UI... fingers crossed though, haven't researched that yet)

Sorry if I've missed anything, I have loads of ideas, but little to no time to execute ;)

Cheers

**Comment by radusuciu** [comment]
@Saiyato your docker containers look great, and the general idea sounds much easier to maintain than a plugin. 

Are you planning on dockerizing volumio as well, or running docker on the volumio host, or separate hosts?

**Comment by Saiyato** [comment]
Hi @radusuciu ,

I have no intent to dockerize Volumio, this has way more implications that I can currently comprehend (or maintain) :wink:
The containers have been downloaded quite a few times and only a few issues were registered, so I guess they're quite stable.

I also worked on these new plugins, separating the server and client parts:
https://github.com/Saiyato/volumio-snapserver-plugin
https://github.com/Saiyato/volumio-snapclient-plugin
